ambry-account/src/integration-test/java/com/github/ambry/account/MySqlAccountServiceIntegrationTest.java

```diff
@@ -739,6 +745,48 @@ public void testAddAndGetDatasets() throws Exception {
             testContainer.getName(), DATASET_NAME);
 
     assertEquals("Mistmatch in dataset read from db", dataset, datasetFromMysql);
+
+    //update the retention count only.
+    int newRetentionCount = 10;
+    Dataset datasetForUpdate =
+        new DatasetBuilder(testAccount.getName(), testContainer.getName(), DATASET_NAME).setRetentionCount(10).build();
+    mySqlAccountStore.updateDataset(testAccount.getId(), testContainer.getId(), datasetForUpdate);
+    datasetFromMysql = mySqlAccountStore.getDataset(testAccount.getId(), testContainer.getId(), testAccount.getName(),
+        testContainer.getName(), DATASET_NAME);
+    assertEquals("Mismatch in updated retention count", (Integer) newRetentionCount,
+        datasetFromMysql.getRetentionCount());
+
+    //update the expiration time.
+    long expirationTimeInMs = Utils.addSecondsToEpochTime(System.currentTimeMillis(), 30);
+    datasetForUpdate =
+        new DatasetBuilder(testAccount.getName(), testContainer.getName(), DATASET_NAME).setExpirationTimeMs(
+            expirationTimeInMs).build();
+    mySqlAccountStore.updateDataset(testAccount.getId(), testContainer.getId(), datasetForUpdate);
+    datasetFromMysql = mySqlAccountStore.getDataset(testAccount.getId(), testContainer.getId(), testAccount.getName(),
+        testContainer.getName(), DATASET_NAME);
+    assertEquals("Mismatch in updated expirationTimeInMst", (Long) expirationTimeInMs,
+        datasetFromMysql.getExpirationTimeMs());
```

Comment:
could you please add another test where the delete_ts is not -1 when created, and later we update to -1.

Reply:
Updated.